niworkflows/interfaces/images.py

```diff
@@ -670,25 +670,38 @@ class SignalExtraction(SimpleInterface):
     output_spec = SignalExtractionOutputSpec
 
     def _run_interface(self, runtime):
+        img = nb.load(self.inputs.in_file)
         mask_imgs = [nb.load(fname) for fname in self.inputs.label_files]
-        if len(mask_imgs) == 1:
+        multi_index_mask = False
+        if len(mask_imgs) == 1 and len(mask_imgs[0].shape) == 4:
             mask_imgs = nb.four_to_three(mask_imgs[0])
 
-        masks = [mask_img.get_data().astype(np.bool) for mask_img in mask_imgs]
+        for i, mask in enumerate(mask_imgs):
+            if img.shape != mask.shape:
+                mask_imgs[i] = nli.resample_to_img(
+                    mask, img, interpolation='nearest')
 
-        n_masks = len(masks)
+        if len(mask_imgs) > 1:
+            masks = [mask_img.get_data().astype(np.bool)
```

Comment:
Let's add an input with 0.5 by default, wdyt about prob_thres?
```suggestion
            masks = [mask_img.get_data() > self.inputs.prob_thres
```